File: frontend/src/services/clients.ts
import { api } from "./api";
import type { ClientProfile, ClientProfileUpdateInput } from "../types";

export function getMyProfile(): Promise<ClientProfile> {
  return api.get<ClientProfile>("/clients/me");
}

export function updateMyProfile(payload: ClientProfileUpdateInput): Promise<ClientProfile> {
  return api.put<ClientProfile>("/clients/me", payload);
}

File: backend/app/api/v1/clients.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.dependencies.auth import require_client
from app.models.user import User
from app.schemas.client import ClientProfileOut, ClientProfileUpdate
from app.services import client_service

router = APIRouter(prefix="/clients", tags=["clients"], dependencies=[Depends(require_client)])


@router.get("/me", response_model=ClientProfileOut)
def get_my_profile(db: Session = Depends(get_db), user: User = Depends(require_client)) -> ClientProfileOut:
    return client_service.get_my_profile(db, user)


@router.put("/me", response_model=ClientProfileOut)
def update_my_profile(
    payload: ClientProfileUpdate,
    db: Session = Depends(get_db),
    user: User = Depends(require_client),
) -> ClientProfileOut:
    return client_service.update_my_profile(db, user, payload)
